### PR TITLE
feat(ci): create setup-nexus-go action

### PR DESCRIPTION
**Pull request by jonathankao97**
This PR will allow developers to use the setup-nexus-go composite action within their workflows. See the following example.

  - uses: actions/setup-go@v3
    with:
      go-version: '^1.13.1'
  - name: Set up setup-nexus-go
    uses: sportsball-ai/actions-hub/setup-nexus-go@setup-oe
    with:
      endpoint:
      login: ${{ secrets.some_login }}
      password: ${{ secrets.some_password }}
  - name: install proprietary package
    run: |
      go mod init github.com/jonathankao97/test
      go get github.com/proprietary@version

### CODE REVIEW

File: setup-oe/action.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      with:
+        go-version: '^1.18.3'
+    - run: |
+        export GONOSUMDB=github.com/sportsball-ai
+        export GOPROXY=https://${{ inputs.endpoint }}/repository/go
+        export GOSUMDB="sum.golang.org https://${{ inputs.endpoint }}/repository/gosumdb"
+        rm -f ~/.netrc
+        echo machine ${{ inputs.endpoint }} >> ~/.netrc
+        echo login ${{ inputs.login }} >> ~/.netrc
+        echo password ${{  inputs.password }} >> ~/.netrc
```
Comment (beiriannydd) [inline setup-oe/action.yml]: is there any way to make sure that these are secrets?
is there documentation?

**Review by beiriannydd — CHANGES_REQUESTED** [review]
I really like the idea of this, it is actually a fairly general case "setup go" action.  Maybe we should call it that instead of setup-oe.  I really want us to make sure we keep our documentation up to date.  Is there any way that we can have tests for actions?

File: setup-oe/action.yml (hunk 1 above)
Comment (jonathankao97) [reply setup-oe/action.yml]: All workflow runs still occur in the environment where the composite action is used. If they are stored as a secret in that workflow, they will be treated as a secret throughout

**Comment by jonathankao97** [comment]
@beiriannydd i see what you mean by this action being a generic setup-go action. Should consumers of this action first pull setup-go and then manually run go install with the correct endpoint?

File: setup-nexus-go/action.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    required: true
+runs:
+  using: composite
+  steps:
+    - run: |
+        echo GONOSUMDB=github.com/sportsball-ai >> $GITHUB_ENV
+        echo GOPROXY=https://${{ inputs.endpoint }}/repository/go >> $GITHUB_ENV
+        echo GOSUMDB="sum.golang.org https://${{ inputs.endpoint }}/repository/gosumdb" >> $GITHUB_ENV
+        rm -f ~/.netrc
+        echo machine ${{ inputs.endpoint }} >> ~/.netrc
```
Comment (raven-txm) [inline setup-nexus-go/action.yml]: Instead of being multiple lines of echo, can we just use a bash heredoc?
https://mywiki.wooledge.org/HereDocument

**Review by raven-txm — CHANGES_REQUESTED** [review]

**Review by raven-txm — APPROVED** [review]

**Pull request merged**